hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncConnectionImpl.java

```diff
@@ -186,6 +183,33 @@ private void spawnRenewalChore(final UserGroupInformation user) {
     authService.scheduleChore(AuthUtil.getAuthRenewalChore(user));
   }
 
+  /**
+   * Get a unique key for the rpc stub to the given server.
+   */
+  private String getStubKey(String serviceName, ServerName serverName) throws UnknownHostException {
+    // Sometimes, servers go down and they come back up with the same hostname but a different
+    // IP address. Force a resolution of the hostname by trying to instantiate an
+    // InetSocketAddress, and this way we will rightfully get a new stubKey.
+    // Also, include the hostname in the key so as to take care of those cases where the
+    // DNS name is different but IP address remains the same.
+    String hostname = serverName.getHostname();
+    int port = serverName.getPort();
+    // We used to ignore when the address was unresolvable but that makes no sense. It
```

Comment:
So we should use the ServerName here directly to make the key instead? 

I can do that. Let me make the change.

Reply:
I'm not sure whether we really need to identify different region servers here. What is the problem if we just use host and port as the stub key here? It will not be a problem if we will resolve it when we actually connecting the remote side? Thoughts?